backend/wallet/services/transaction.py

```diff
@@ -90,10 +97,15 @@ def process_incoming_transaction(
         if general_v0.to_subaddress:
             receiver_subaddr = general_v0.to_subaddress.hex()
             receiver_id = get_account_id_from_subaddr(receiver_subaddr)
+        else:
+            receiver_id = get_account(account_name=INVENTORY_ACCOUNT_NAME).id
```

Comment:
don't understand here, if we set receiver_id, then the `if receiver_id is None:` probably won't work as expected

Reply:
This is the case when we receive money with no subaddress, which means it goes to our main inventory account

Reply:
I understand it is for no subaddress case, but there is `if receiver_id is None:` check in the following code, what is that for after you added here.

Reply:
If general_v0.to_subaddress is None, it means it was sent to our on-chain account without any subaddress defined, which means we want to send it to our inventory account, which is why I set receiver_id. 

Later I check if receiver_id is None, which means there was a subaddress provided, but receiver ID lookup failed which means the subaddress the sender gave is invalid. In that case we send back a refund, which is handled below.